lms/djangoapps/learner_recommendations/views.py

```diff
@@ -137,60 +137,120 @@ class CrossProductRecommendationsView(APIView):
     """
 
     def _empty_response(self):
+        """Helper for sending an empty response"""
+
         return Response({"courses": []}, status=200)
 
-    def get(self, request, course_id):
+    def _amplitude_recommendations(self, user, user_country_code):
         """
-        Returns cross product recommendation courses
+        Helper for getting amplitude recommendations
+        """
+
+        fallback_recommendations = settings.GENERAL_RECOMMENDATIONS[0:4]
+
+        try:
+            _, _, course_keys = get_amplitude_course_recommendations(
+                user.id, settings.DASHBOARD_AMPLITUDE_RECOMMENDATION_ID
+            )
+        except Exception as ex:  # pylint: disable=broad-except
+            log.warning(f"Cannot get recommendations from Amplitude: {ex}")
+            return fallback_recommendations
+
+        if not course_keys:
+            return fallback_recommendations
+
+        filtered_courses = get_filtered_discovery_course_data(
+            course_keys, user_country_code, user, amplitude_course_filters=True
+        )
+
+        return filtered_courses if len(filtered_courses) == 4 else fallback_recommendations
```

Comment:
Why are we getting rid of Amplitude recommendations if we don't get 4 filtered recs? The current panel shows the fallback recommendations only if we are left with zero recommendations.